resources/lib/comm.py

```diff
@@ -0,0 +1,143 @@
+import classes
+import config
+import json
+
+from aussieaddonscommon import session
+
+
+def fetch_url(url, headers={}):
+    """
+    Use custom session to grab URL and return the text
+    """
+    with session.Session(force_tlsv1=True) as sess:
+        res = sess.get(url, headers=headers)
+        return res.text
+
+
+def list_series():
+    """
+    Create and return list of series objects
+    """
+    res = fetch_url(config.TVSERIES_URL)
+    data = json.loads(res)
+    listing = []
+    for show in data['items']:
+        for season in reversed(show['containsSeason']):
+            s = classes.series()
+            s.multi_season = len(show['containsSeason']) > 1
+            s.season_slug = season['slug']
+            s.series_name = show['name']
+            s.season_name = season['name']
+            s.series_slug = season['partOfSeries']['slug']
+            s.fanart = show['image']['sizes']['w1920']
+            s.thumb = season['image']['sizes']['w480']
+            s.genre = season['genre']['name']
+            s.title = s.get_title()
+            listing.append(s)
+    return listing
+
+
+def list_genres():
+    """
+    Create and return list of genre objects
+    """
+    res = fetch_url(config.GENRES_URL)
+    data = json.loads(res)
+    listing = []
+    for genre in data['items']:
+        g = classes.genre()
+        g.fanart = (genre['image']['sizes']['w1920'])
+        g.thumb = (genre['image']['sizes']['w480'])
+        g.genre_slug = genre['slug']
+        g.title = genre['name']
+        listing.append(g)
+    return listing
+
+
+def list_episodes(params):
+    """
+    Create and return list of episode objects
+    """
+    url = config.EPISODEQUERY_URL.format(
+        params['series_slug'], params['season_slug'])
+    res = fetch_url(url)
+    data = json.loads(res)
+    listing = []
+    for section in data['items']:
+        # filter extras etc for most shows.
+        if section.get('callToAction'):
+            if (section['callToAction']['link']['type'] not in
+               ['episode-index', 'external']):
+                continue
+        for episode in section['items']:
+            # filter extras again as some show are unable to be filtered at the
+            # previous step
+            if not episode.get('episodeNumber'):
+                continue
+            # make sure season numbers match, some shows return all seasons.
+            if episode['partOfSeason'].get('slug') != params['season_slug']:
+                continue
+
+            e = classes.episode()
+            e.episode_no = str(episode['episodeNumber'])
+            e.thumb = episode['image']['sizes']['w480']
+            e.fanart = data['tvSeries']['image']['sizes']['w1920']
+            e.episode_name = episode['name'].encode('utf8')
+            e.title = e.get_title()
+            e.desc = episode['description']
+            e.duration = episode['video']['duration']//1000
```

Comment:
is // right?

Reply:
It's not wrong but unnecessary for python 2, we get an int either way.

Reply:
OK, I haven't seen it before. Is it a python3 requirement? If so, then I'm happy for it to stay

Reply:
not a requirement for python 3 but the listitem wants an int and in python 3 the / operator returns a float, // is floor division and returns an int. I'll just make it a / for now. 

This happened recently though: https://github.com/xbmc/xbmc/pull/12705 and will eventually make its way into master.

Reply:
Nice to be as compatible with Py3 as possible. You should leave it. Nice find though